packages/medusa/src/api/routes/admin/draft-orders/create-draft-order.js

```diff
@@ -0,0 +1,77 @@
+import { MedusaError, Validator } from "medusa-core-utils"
+import { defaultFields, defaultRelations } from "."
+
+export default async (req, res) => {
+  const schema = Validator.object().keys({
+    status: Validator.string()
+      .valid("open", "awaiting", "completed")
+      .optional(),
+    email: Validator.string()
+      .email()
+      .required(),
+    billing_address: Validator.address().optional(),
+    shipping_address: Validator.address().optional(),
+    billing_address_id: Validator.string().optional(),
+    shipping_address_id: Validator.string().optional(),
+    items: Validator.array()
+      .items({
+        variant_id: Validator.string()
+          .optional()
+          .allow(""),
+        unit_price: Validator.number().optional(),
+        title: Validator.string()
+          .optional()
+          .allow(""),
+        quantity: Validator.number().required(),
+        metadata: Validator.object().default({}),
+      })
+      .required(),
+    region_id: Validator.string().required(),
+    discounts: Validator.array().optional(),
+    customer_id: Validator.string().optional(),
+    customer: Validator.string().optional(),
+    shipping_methods: Validator.array()
+      .items({
+        option_id: Validator.string().required(),
+        data: Validator.object().optional(),
+        price: Validator.number()
+          .integer()
+          .integer()
+          .allow(0)
+          .optional(),
+      })
+      .optional(),
+    metadata: Validator.object().optional(),
+    requires_shipping: Validator.boolean().default(true),
```

Comment:
Thinking that is it is the backend's responsibility to figure out if the order requires shipping or not.

If there are items in the cart that are inventory tracked then they should have a shipping method that can fulfill them. If we don't have a shipping method that can fulfill the items then we will have trouble with fulfillment once the draft order is promoted to an order. 

In cases where the store operator wants to just hand over the goods (i.e. in person purchase) they should use the manual fulfillment provider to do so.

Reply:
This may introduce the necessity for a change in the way shipping methods work. For example, if you have a store where people from time to time walk by and purchase something or if you sell from your office you may want to have a shipping method called "In store fulfillment", this shipping method should not be shown on the website but only be available within Admin.